tools/clang/unittests/HLSL/WEXAdapter.h

```diff
@@ -0,0 +1,137 @@
+#define VERIFY_ARE_EQUAL_3(A, B, msg) EXPECT_EQ(A, B) << msg
+#define VERIFY_ARE_EQUAL(...) CALL_MACRO_N(VERIFY_ARE_EQUAL_, __VA_ARGS__)
+
+
+// Macros to convert TAEF macros to gtest equivalents
+#define VERIFY_IS_TRUE EXPECT_TRUE
+#define VERIFY_FAILED(expr) EXPECT_FALSE(SUCCEEDED(expr))
+#define VERIFY_IS_FALSE EXPECT_FALSE
+#define VERIFY_IS_NULL(exp) EXPECT_EQ(nullptr, (exp))
+#define VERIFY_IS_NOT_NULL(exp) EXPECT_NE(nullptr, (exp))
+#define VERIFY_FAIL ADD_FAILURE
+#define VERIFY_ARE_NOT_EQUAL EXPECT_NE
+#define VERIFY_WIN32_BOOL_SUCCEEDED EXPECT_TRUE
+
+#define TEST_CLASS_SETUP(method) bool method(); virtual void SetUp() { EXPECT_TRUE(method()); }
+#define TEST_CLASS_CLEANUP(method) bool method(); virtual void TearDown() { EXPECT_TRUE(method()); }
+#define BEGIN_TEST_CLASS(test)
+#define TEST_CLASS_PROPERTY(str1, str2)
+#define TEST_METHOD_PROPERTY(str1, str2)
+#define END_TEST_CLASS()
+#define TEST_METHOD(method)
+#define BEGIN_TEST_METHOD(method)
+#define END_TEST_METHOD()
+
+// gtest lacks any module setup/cleanup. These functions are called by the main() function
+// before and after tests are run. This approximates the behavior.
+bool moduleSetup();
+bool moduleTeardown();
+#define MODULE_SETUP(method) bool method(); bool moduleSetup() { return method(); }
+#define MODULE_CLEANUP(method) bool method(); bool moduleTeardown() { return method(); }
+
+// No need to expand env vars on Unix platforms, so convert the slashes instead.
+inline DWORD ExpandEnvironmentStringsW(_In_ LPCWSTR lpSrc,_Out_opt_ LPWSTR lpDst, _In_ DWORD nSize) {
+  unsigned i;
+  bool wasSlash = false;
+  for (i = 0; i < nSize && *lpSrc; i++, lpSrc++) {
+    if(*lpSrc == L'\\' || *lpSrc == L'/') {
+      if (!wasSlash)
+        *lpDst++ = L'/';
+      wasSlash = true;
+    } else {
+      *lpDst++ = *lpSrc;
+      wasSlash = false;
+    }
+  }
+  *lpDst = L'\0';
+  return i;
+}
+
+typedef struct _LIST_ENTRY {
+  struct _LIST_ENTRY *Flink;
+  struct _LIST_ENTRY *Blink;
+} LIST_ENTRY, *PLIST_ENTRY, PRLIST_ENTRY;
+
+// Minimal implementation of the WEX namespace functions and classes
+// To either stub out or approximate behavior
+namespace WEX {
+namespace Common {
+  class String : public std::wstring {
+  public:
+    size_t GetLength() {return length();}
+    bool IsEmpty() {return empty();}
+    int CompareNoCase(std::wstring str) const {return compare(str);}
+    operator const wchar_t *() {return c_str();}
+    const wchar_t *GetBuffer() {return *this;}
+    wchar_t *Format(const wchar_t *fmt, ...) {
+      static wchar_t msg[512];
+      va_list args;
+      va_start(args, fmt);
+      vswprintf(msg, 512, fmt, args);
+      va_end(args);
+      return msg;
+    }
+  };
+} // Common namespace
+namespace TestExecution {
+  enum class VerifyOutputSettings {
+    LogOnlyFailures
+  };
+  class SetVerifyOutput {
+  public:
+    SetVerifyOutput(VerifyOutputSettings) {}
+  };
+  class DisableVerifyExceptions {
+  public:
+    DisableVerifyExceptions() {}
```

Comment:
According to the [doc](https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/verify):
if you would like to perform a series of verifications in a row rather than having the test abort on the first verification failure, you can use the DisableVerifyExceptions class. The object's lifetime controls the amount of time that exceptions are disabled.

So we are basically disabling this feature on Unix, which is reasonable IMO.